packages/eth/src/websocket/connection.rs

```diff
@@ -1,154 +1,173 @@
+        let tx_receipt = self
+            .provider
+            .get_transaction_receipt(tx_hash.into())
+            .await?;
 
         Self::convert_to_tx_response(tx_receipt)
     }
 
     async fn submit_l2_state(&self, state_data: Vec<u8>) -> Result<[u8; 32]> {
-        let blob_pool_signer = if let Some(blob_pool_signer) = &self.blob_signer {
-            blob_pool_signer
-        } else {
-            return Err(Error::Other("blob pool signer not configured".to_string()));
+        let (blob_provider, blob_address) = match (&self.blob_provider, &self.blob_address) {
+            (Some(provider), Some(address)) => (provider, address),
+            _ => return Err(Error::Other("blob pool signer not configured".to_string())),
         };
 
-        let sidecar = BlobSidecar::new(state_data).map_err(|e| Error::Other(e.to_string()))?;
-        let blob_tx = self
-            .prepare_blob_tx(
-                sidecar.versioned_hashes(),
-                blob_pool_signer.address(),
-                blob_pool_signer.chain_id(),
-            )
-            .await?;
-
-        let tx_encoder = BlobTransactionEncoder::new(blob_tx, sidecar);
-        let (tx_hash, raw_tx) = tx_encoder.raw_signed_w_sidecar(blob_pool_signer).await?;
+        let blob_tx = self.prepare_blob_tx(&state_data, *blob_address).await?;
 
-        self.provider.send_raw_transaction(raw_tx.into()).await?;
+        let tx = blob_provider.send_transaction(blob_tx).await?;
 
-        Ok(tx_hash.to_fixed_bytes())
+        Ok(tx.tx_hash().0)
     }
 
     #[cfg(feature = "test-helpers")]
     async fn finalized(&self, block: ValidatedFuelBlock) -> Result<bool> {
         Ok(self
             .contract
-            .finalized(block.hash(), block.height().into())
+            .finalized(block.hash().into(), U256::from(block.height()))
             .call()
-            .await?)
+            .await?
+            ._0)
     }
 
     #[cfg(feature = "test-helpers")]
     async fn block_hash_at_commit_height(&self, commit_height: u32) -> Result<[u8; 32]> {
         Ok(self
             .contract
-            .block_hash_at_commit(commit_height.into())
+            .blockHashAtCommit(U256::from(commit_height))
             .call()
-            .await?)
+            .await?
+            ._0
+            .into())
     }
 }
 
 impl WsConnection {
     pub async fn connect(
-        url: &Url,
+        url: Url,
         contract_address: Address,
         main_signer: AwsSigner,
         blob_signer: Option<AwsSigner>,
     ) -> Result<Self> {
-        let provider = Provider::<Ws>::connect(url.to_string()).await?;
-
         let address = main_signer.address();
 
-        let signer = SignerMiddleware::new(provider.clone(), main_signer.clone());
+        let ws = WsConnect::new(url);
+        let provider = Self::provider_with_signer(ws.clone(), main_signer).await?;
+
+        let (blob_provider, blob_address) = if let Some(signer) = blob_signer {
```

Comment:
rather than keeping Provider and address optional request them always and turn of the functionality with some flag